javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -42,13 +42,23 @@
     private static ClientMethodMapper instance = new ClientMethodMapper();
     private Map<Operation, List<ClientMethod>> parsed = new HashMap<>();
 
-    private ClientMethodMapper() {
+    protected ClientMethodMapper() {
     }
 
     public static ClientMethodMapper getInstance() {
         return instance;
     }
 
+    public void addModelsTo(List<ClientModel> clientModels) {
```

Comment:
Seems better just have a `getOptionsModels` and return the List of `ClientModel`s, easier to understand what it for, IMHO. Also please add javadoc.

Reply:
I'm just trying to follow addImportsTo. :-) In fact, passing what existed already may give this class a chance to check for duplicates (as you suggested in other comment). If people want to be smarter in renaming the newly generated model, this is a chance for doing so.

Reply:
Sure. I've no strong opinion here.